### ISSUE TITLE
Snowflake: Missing ALTER DYNAMIC TABLE support

### ISSUE THREAD

**Issue by gws** [sqlfluff/sqlfluff#7171]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The Snowflake dialect appears to be missing support for `ALTER DYNAMIC TABLE` statements, which causes parse errors when these statements are used standalone or within other statements like `CREATE TASK`.

### Expected Behaviour

The following snippets should be parseable (if perhaps not correctly formatted):

```sql
ALTER DYNAMIC TABLE foo REFRESH;
```

```sql
CREATE TASK mytask
AS
ALTER DYNAMIC TABLE foo REFRESH;
```

### Observed Behaviour

The below snippets cause parse errors:

```sql
ALTER DYNAMIC TABLE foo REFRESH;
```

```sql
CREATE TASK mytask
AS
ALTER DYNAMIC TABLE foo REFRESH;
```

### How to reproduce

```sh
echo 'ALTER DYNAMIC TABLE foo REFRESH;' | sqlfluff lint -d snowflake -
echo 'CREATE TASK mytask AS ALTER DYNAMIC TABLE foo REFRESH;' | sqlfluff lint -d snowflake -
```

### Dialect

Snowflake

### Version

sqlfluff, version 3.4.2

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
